centos-ci/clang-analysis/jenkins-job.py

```diff
@@ -0,0 +1,60 @@
+#
+# from: https://raw.githubusercontent.com/kbsingh/centos-ci-scripts/master/build_python_script.py
+#
+# This script uses the Duffy node management api to get fresh machines to run
+# your CI tests on. Once allocated you will be able to ssh into that machine
+# as the root user and setup the environ
+#
+# XXX: You need to add your own api key below, and also set the right cmd= line
+#      needed to run the tests
+#
+# Please note, this is a basic script, there is no error handling and there are
+# no real tests for any exceptions. Patches welcome!
+
+import json, urllib, subprocess, sys, os
+
+print ""
+print "This test is part of the Gluster Jenkins jobs that can be found on GitHub:"
+print " - https://github.com/gluster/glusterfs/tree/master/extras"
+print ""
+
+url_base = "http://admin.ci.centos.org:8080"
+ver = "7"
+arch = "x86_64"
+count = 1
+script_url = os.getenv("TEST_SCRIPT")
+burl = os.getenv("BUILD_URL")
+burl += 'consoleFull'
+
+# read the API key for Duffy from the ~/duffy.key file
+fo = open("/home/gluster/duffy.key")
+api = fo.read().strip()
+fo.close()
+
+# build the URL to request the system(s)
+get_nodes_url = "%s/Node/get?key=%s&ver=%s&arch=%s&count=%s" % (url_base, api, ver, arch,count)
+
+# request the system
+dat = urllib.urlopen(get_nodes_url).read()
+b = json.loads(dat)
+cmd = """ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
+      yum -y install curl && curl %s | bash - '""" % (b['hosts'][0], script_url)
+rtn_code=subprocess.call(cmd, shell=True)
+
+# Update Gerrit with the success/failure status
+if rtn_code == 0:
+    v = "+1"
+    verdict = "SUCCESS"
+else:
+    v = "-1"
+    verdict = "FAILED"
+
+cmd = "ssh build@review.gluster.org gerrit review --message \"\'%s : %s\'\" --project=glusterfs --label Clang-analysis=\"%s\" %s" % (burl, verdict, v, os.getenv("GIT_COMMIT"))
```

Comment:
On Fri, Apr 01, 2016 at 06:50:48AM -0700, Kaushal M wrote:
> > +# request the system
> > +dat = urllib.urlopen(get_nodes_url).read()
> > +b = json.loads(dat)
> > +cmd = """ssh -t -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@%s '
> > +      yum -y install curl && curl %s | bash - '""" % (b['hosts'][0], script_url)
> > +rtn_code=subprocess.call(cmd, shell=True)
> > +
> > +# Update Gerrit with the success/failure status
> > +if rtn_code == 0:
> > +    v = "+1"
> > +    verdict = "SUCCESS"
> > +else:
> > +    v = "-1"
> > +    verdict = "FAILED"
> > +
> > +cmd = "ssh build@review.gluster.org gerrit review --message \"\'%s : %s\'\" --project=glusterfs --label Clang-analysis=\"%s\" %s" % (burl, verdict, v, os.getenv("GIT_COMMIT"))
> 
> Use `centos-ci` as the user.
> 
> @nixpanic is the public key you provided when I created the bot-account from this machine?

That is the public key that the CentOS CI uses to connect to
review.gluster.org to get the patch triggers. I'll add the private key
to the gluster user on slave01.ci.centos.org.